### ISSUE TITLE
Questions for Custom LD Reference Data

### ISSUE THREAD

**Issue by maegsul** [gusevlab/fusion_twas#28]
Hi,

First of all, thanks once again @sashagusev for developing & maintaining this great tool, I have been using it especially past year quite a lot. Lately I stopped using the main LD reference data available [here](http://gusevlab.org/projects/fusion/#installation) on FUSION website and decided to create a custom LD reference data for following reasons:

- I've switched to hg38/GRCh38 for QTL/TWAS analyses (provided LD reference data is on hg19/GRCh37).
- Provided LD reference data contains n=489 samples, however n=99 of these are Finnish samples and I plan to exclude these (as a side note here, actually in 1KG there are n=14 additional EUR non-Finnish samples that are missing in this n=489 provided LD reference data).
- My QTL mapping is based on a very large mainly EU-ancestry WGS dataset, and a total of ~14M variants (all MAF > 1% and genotype & variant QCed) are actually tested. The available LD reference data only contains 1,190,321 variants (98.8% of these variants are matching & tested in my QTL catalogues), so I want to account for the remaining tested (and for TWAS -> testable) variants.

I looked into different scenarios for creating a custom LD reference data, I also followed a few issues/comments already mentioned in this repository, however I still have several questions.

On 1KG ftp server, for the same 1KG samples, you can find [one "old style" GRCh38 1KG VCF](http://ftp.1000genomes.ebi.ac.uk/vol1/ftp/data_collections/1000_genomes_project/release/20190312_biallelic_SNV_and_INDEL/) and [one "new style" GRCh38 1KG VCF](http://ftp.1000genomes.ebi.ac.uk/vol1/ftp/data_collections/1000G_2504_high_coverage/working/20190425_NYGC_GATK/). If I am not wrong, the former is based on old low-coverage WGS + WES data (guess provided LD reference data is based on this as well?) and the latter is recent PCR-free high-coverage WGS (resequenced by NYGC).

The old style 1KG VCF have only **phased** (with shapeit2) genotypes and contains only GT information in the FORMAT field. The new style 1KG VCF have all modern FORMAT field values: GT, DP, AB, AD, GQ; therefore it is possible to do genotype QC filters such as assigning genotypes missing if GQ < 20 or if allele reads are not balanced. 

The new style 1KG VCF overlaps much better with my WGS data in terms of list of variants (97.68%) , though the old style VCF is also still good (87.16%). 

My questions here:

1) For custom LD reference data (generated only for TWAS), is it recommended to prefer phased data over unphased data? In this case, as a reminder, the old style VCF is actually phased. However, I think for TWAS this does not matter as it accepts PLINK binary files and at this level phase information of genotypes are already lost. Is this true?

2) Do you recommend doing genotype & variant QC on the LD reference data? In my case I only filtered out the variants tagged by GATK VQSR pipeline for the new style 1KG VCF. However, I also consider genotype filters mentioned above, and maybe also HWE (as you indicated in this [comment](https://github.com/gusevlab/fusion_twas/issues/7#issuecomment-484518152), however I don't understand why HapMap3 restriction was applied, can you please elaborate more on this decision? Maybe this is not required for the new PCR-free high-coverage WGS VCF, as no imputation needed different than last time?)

3) Should the genotyping rate of LD reference data be always 100%, meaning that none of the genotypes are missing? This is the case in the provided LD reference data, and also for the old style 1KG VCF I've mentioned, but not for the new one. I am not sure if FUSION is compatible with these missing genotypes in LD reference data? As a side note, the new VCF currently has only a small portion of missingness (total genotyping rate 99.96%).

4) Should I remove monomorphic sites in the LD reference data? (even if they match with my list of tested QTL variants). Of note, currently about 10% of these variants matching between LD reference data <-> my testable QTL variant list are with MAC = 0.

Thanks in advance!
Fahri

**Comment by maegsul** [comment]
Just an additional note: for now, I chose to move forward with the more recent QCed unphased high-covarage LD reference data (not filtering for monomorphic variants). When I run FUSION.assoc_test.R with this data it throws this error:

"Error in solve.default(cur.LD[!cur.miss, !cur.miss] + 0.1 * diag(sum(!cur.miss))) :
system is computationally singular: reciprocal condition number = 0
Calls: solve -> solve.default
Execution halted"

However, when I do --mac 1 in PLINK to remove monomorphic variants in the LD reference data, FUSION.assoc_test.R works without a problem (even I use a LD reference data that is unphased and that have missing genotypes after genotype & variant QC, but still a total genotyping rate of about 99%). I still would love to hear about your thoughts on this of course.

**Comment by sashagusev** [comment]
Hi, thanks for raising this issue. A few responses below:

> For custom LD reference data (generated only for TWAS), is it recommended to prefer phased data over unphased data? In this case, as a reminder, the old style VCF is actually phased. However, I think for TWAS this does not matter as it accepts PLINK binary files and at this level phase information of genotypes are already lost. Is this true?

Correct, TWAS takes in PLINK binary files which do not reflect the phasing and TWAS does not use any phasing information.

> Do you recommend doing genotype & variant QC on the LD reference data? In my case I only filtered out the variants tagged by GATK VQSR pipeline for the new style 1KG VCF. However, I also consider genotype filters mentioned above, and maybe also HWE (as you indicated in this comment, however I don't understand why HapMap3 restriction was applied, can you please elaborate more on this decision? Maybe this is not required for the new PCR-free high-coverage WGS VCF, as no imputation needed different than last time?)

Yes, we recommend performing basic variant QC (MAF>1%; missingness <1%; HWE P-value > 10e-7). At minimum, the software will crash or throw a warning if monomorphic variants are included. We additionally restrict to HapMap3 SNPs because these variants have been shown empirically to capture common GWAS heritability well and to be consistently imputed across most genotype arrays, so this is a conservative way to ensure that SNPs being used are generally high quality. If you are using your own reference data and building your own weights you do not need to impose this restriction (in general we find that it makes little difference in accuracy but increases performance to restrict).

> Should the genotyping rate of LD reference data be always 100%, meaning that none of the genotypes are missing? This is the case in the provided LD reference data, and also for the old style 1KG VCF I've mentioned, but not for the new one. I am not sure if FUSION is compatible with these missing genotypes in LD reference data? As a side note, the new VCF currently has only a small portion of missingness (total genotyping rate 99.96%).

Some amount of missingness is okay (<1%) and FUSION will fill in the missing data based on the variant mean.

> Should I remove monomorphic sites in the LD reference data? (even if they match with my list of tested QTL variants). Of note, currently about 10% of these variants matching between LD reference data <-> my testable QTL variant list are with MAC = 0.

Yes, monomorphic sites violate the modelling assumptions (normally distributed variants) and can break the analysis so these should be removed.

Hope that helps!

**Comment by maegsul** [comment]
Thank you very much @sashagusev ! Happy that the methods for building weights & LD reference data are in line mostly.

An update: I actually switched to the "old style" GRCh38 1KG VCF LD Panel that I referred to earlier, because the "new style" one gave me very unexpected heritability estimates (even for some of the genes that I am very confident that heritability should be high). Overall, the old-style GRCh38 custom LD panel outputted the ~30% of features (of total n=~25K genes) with hsq_p < 0.01, but this ratio was only 3% when this new-style custom LD panel used. Maybe I did something wrong somewhere, but that caused me to change my previous preference for the most recent 1KG GRCh38 data.

I am now happy with the performance of this "old-style" custom LD panel (GRCh38 1KG EUR n=404 phased) I've prepared, and I can provide more information if needed. It contains variants that are non-monomorphic, non-missing (naturally, as it is phased) and with HWE P > 1E-6. Total autosomal variant number is about ~21.6M (as I didn't remove MAF < 1% variants in the custom LD reference panel).
